subprojects/core/src/main/java/org/gradle/caching/internal/tasks/TarTaskOutputPacker.java

```diff
@@ -312,15 +328,23 @@ private void unpackPropertyEntry(ResolvedTaskOutputFilePropertySpec propertySpec
             outputFile = new File(propertyRoot, childPath);
         }
 
+        String internedPath = stringInterner.intern(outputFile.getAbsolutePath());
+        RelativePath relativePath = root ? RelativePath.EMPTY_ROOT : RelativePath.parse(!isDirEntry, childPath);
         if (isDirEntry) {
             FileUtils.forceMkdir(outputFile);
+            if (!root) {
+                fileSnapshots.put(propertyName, new DirectoryFileSnapshot(internedPath, relativePath, false));
```

Comment:
I don't see overlapping outputs when I run the task after it has loaded from cache. The tests now cover this case, too. Why do you think this is a problem?